src/java.base/share/classes/sun/security/util/KeyUtil.java

```diff
@@ -174,6 +174,22 @@ public static final int getKeySize(AlgorithmParameters parameters) {
         return -1;
     }
 
+    /*
+     If the key is a sub-algorithm of a larger group of algorithms, this method
+     will return that sub-algorithm.  For example, key.getAlgorithm() returns
+     "EdDSA", but the underlying key maybe "Ed448".  For
+     DisabledAlgorithmConstraints (DAC), this distinction is important.
+     "EdDSA" means all curves for DAC, but when using it with
+     KeyPairGenerator, EdDSA means Ed25519.
+     */
+    public static String getAlgorithm(Key key) {
+        return switch (key) {
+            case EdECKey ed -> ed.getParams().getName();
+            case XECKey xe -> ((NamedParameterSpec) xe.getParams()).getName();
+            default -> key.getAlgorithm();
```

Comment:
Do you also want to add cases for ML-KEM and ML-DSA keys?

Reply:
@wangweij is planning on name usage for those.  I'm focusing on these older curves.

Reply:
They are already defined. I think you just want to add something like:

```
If (key.getAlgorithm().equals("ML-KEM") || key.getAlgorithm().equals("ML-DSA")) {
   return ((NamedParameterSpec) key.getParams()).getName();
}
```

Not urgent, but useful if one of these algorithms were to weaken or be broken for some reason.

Reply:
Or what about this?
```
        if (key instanceof AsymmetricKey ak) {
            if (ak.getParams() instanceof NamedParameterSpec nps) {
                return nps.getName();
            }
        }
        return key.getAlgorithm();
```

Reply:
I stayed away from that because this is likely being backported

Reply:
Ok, well our backporters are usually good about extracting only what is necessary, but if not, can you file another issue to add support for disabling PQC algorithms?

Reply:
I changed my mind and will use the suggestion.